### PR TITLE
Add Amun lending adapters

### PR DESCRIPTION
**Pull request by loki-sama**
Adds Amun lending simple and interactive adapters
Adds tests for adapters

### CODE REVIEW

**Comment by sobolev-igor** [comment]
For our system to work properly, components returned by token adapter should be the same as returned tokens in `withdraw()` function of interactive adapter. So, you need to use function returning yTokens in `withdraw()` or calculate rate to underlying stablecoin in token adapter.

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        address lendingToken = abi.decode(data, (address));
+        require(
+            tokenAmounts[0].token == getUnderlyingStablecoin(lendingToken),
+            "LBIA: should be underling stablecoin"
+        );
+
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = lendingToken;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+        ERC20(tokenAmounts[0].token).safeApprove(lendingToken, amount, "LBIA[1]");
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:65]: please use `safeApproveMax` here

**Comment by loki-sama** [comment]
There is only one lending token: 0x78f9c12e15ec36C2AB1bE0b2e5f79B71A9ECdFC8

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { AmunAddressStorage } from "../../interfaces/AmunAddressStorage.sol";
+
+/**
+ * @title Interactive adapter for AmunBasket.
+ * @dev Implementation of InteractiveAdapter abstract contract.
+ * @author Timo <Timo@amun.com>
+ */
+contract AmunLendingInteractiveAdapter is InteractiveAdapter, ERC20ProtocolAdapter {
+    using SafeERC20 for ERC20;
+    uint8 referralCode = 101;
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:37]:
```suggestion
    uint8 internal constant REFERRAL_CODE = 101;
```

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        tokensToBeWithdrawn[0] = lendingToken;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+        ERC20(tokenAmounts[0].token).safeApproveMax(lendingToken, amount, "LBIA[1]");
+        try
+            AmunLendingToken(lendingToken).create(
+                tokenAmounts[0].token,
+                amount,
+                address(this),
+                0,
+                referralCode
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:72]:
```suggestion
                REFERRAL_CODE
```

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = getUnderlyingStablecoin(lendingToken);
+
+        try
+            AmunLendingToken(lendingToken).redeem(
+                tokensToBeWithdrawn[0],
+                amount,
+                address(this),
+                0,
+                referralCode
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:107]:
```suggestion
                REFERRAL_CODE
```

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    using SafeERC20 for ERC20;
+    uint8 referralCode = 101;
+
+    /**
+     * @notice Deposits tokens to the AmunBasket.
+     * @param tokenAmounts Array of underlying TokenAmounts - TokenAmount struct with
+     * underlying tokens addresses, underlying tokens amounts to be deposited, and amount types.
+     * @param data ABI-encoded additional parameters:
+     *     - lendingToken - AmunBasket address.
+     * @return tokensToBeWithdrawn Array with one element - rebalancing set address.
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:45]:
```suggestion
     * @return tokensToBeWithdrawn Array with one element - AmunBasket address.
```

File: contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            )
+         {} catch Error(string memory reason) {
+            revert(reason);
+        } catch {
+            revert("LBIA: join fail");
+        }
+    }
+
+    /**
+     * @notice Withdraws tokens from the TokenSet.
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLendingtInteractiveAdapter.sol:82]:
```suggestion
     * @notice Withdraws tokens from the AmunBasket.
```

**Review by sobolev-igor — CHANGES_REQUESTED** [review]

**Pull request merged**